docs/user-guide/load-balancer.md

```diff
@@ -69,5 +82,67 @@ information through `kubectl`:
     Endpoints:    10.64.0.4:80,10.64.1.5:80,10.64.2.4:80
     Session Affinity: None
     No events.
+```
 
 The IP address is listed next to `LoadBalancer Ingress`.
+
+## Loss of client source IP for external traffic
+
+Due to the implementation of this feature, the source IP for sessions as seen in the target container will *not be the original source IP* of the client. This is the default behavior as of Kubernetes v1.4. However, starting in v1.4, an optional alpha feature has been added
+that will preserve the client Source IP for GCE/GKE environments. This feature will be phased in for other cloud providers in subsequent releases.
+
+## Annotation to modify the LoadBalancer behavior for preservation of Source IP
+In 1.4, an Alpha feature has been added that changes the behavior of the external LoadBalancer feature.
+
+This feature can be activated by adding the alpha annotation to the metadata section of the Service Configuration file.
+
+```json
+    {
+      "kind": "Service",
+      "apiVersion": "v1",
+      "metadata": {
+        "name": "example-service",
+        "annotations": {
+            "service.alpha.kubernetes.io/external-traffic": "OnlyLocal"
+        }
+      },
+      "spec": {
+        "ports": [{
+          "port": 8765,
+          "targetPort": 9376
+        }],
+        "selector": {
+          "app": "example"
+        },
+        "type": "LoadBalancer"
+      }
+    }
+```
+
+### Alpha Feature Gate for the 'service.alpha.kubernetes.io/external-traffic' annotation
+
+Alpha features are not enabled by default, they must be enabled using the release gate command line flags
+for kube-controller-manager and kube-proxy.
+See [https://github.com/kubernetes/kubernetes/blob/master/docs/proposals/runtimeconfig.md](Runtime feature flags proposal) for more details on feature gate flags.
+
+If this feature is not enabled in your cluster, this service configuration will be rejected.
+
+Also note this feature is not currently implemented for all cloudproviders/environments.
+
+### Caveats and Limitations when preserving source IPs
+
```

Comment:
Still want a note that this does not yet work on nodePorts, so "proxy style" balancers can not use this yet.